src/server/game/Entities/Object/Object.h

```diff
@@ -487,6 +487,7 @@ class WorldObject : public Object, public WorldLocation
         bool IsWithinDistInMap(WorldObject const* obj, float dist2compare, bool is3D = true) const;
         bool IsWithinLOS(float x, float y, float z) const;
         bool IsWithinLOSInMap(WorldObject const* obj) const;
+        virtual Position GetHitSpherePointFor(Position const dest) const;
```

Comment:
I mean have just `GetHitSpherePointFor`, without `virtual`, in `Object` (with the functionality from `Creature`).

Then, wherever you call `GetHitSpherePointFor`, check if it's a player, and if yes, use the `+2.0f` position instead of calling `GetHitSpherePointFor`.